numpy/core/src/multiarray/buffer.c

```diff
@@ -797,7 +786,14 @@ array_getbuffer(PyObject *obj, Py_buffer *view, int flags)
     view->buf = PyArray_DATA(self);
     view->suboffsets = NULL;
     view->itemsize = PyArray_ITEMSIZE(self);
-    view->readonly = !PyArray_ISWRITEABLE(self);
+    /*
+     * Set a requested buffer to readonly also if the array will be readonly
+     * after a deprecation. This jumps the deprecation, but avoiding the
+     * warning is not convenient here and a warning is given if a writeable
+     * buffer is requested. (Warn on write is cleared if writeable is requested)
+     */
```

Comment:
To be clear - the warn on write is already handled above in `PyArray_FailUnlessWriteable`?